google/cloud/storagebatchoperations/v1/samples/storage_batch_operations_job_samples.cc

```diff
@@ -0,0 +1,178 @@
+        client,
+    std::vector<std::string> const& argv) {
+  //! [storage_batch_create_job]
+  [](google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+         client,
+     std::string const& parent, std::string const& job_id) {
+    namespace sbo = google::cloud::storagebatchoperations::v1;
+    sbo::Job job;
+    auto future = client.CreateJob(parent, job, job_id);
+    auto result = future.get();
+    if (!result) throw result.status();
+    std::cout << "Created job: " << result->DebugString() << "\n";
+  }
+  //! [storage_batch_create_job]
+  (std::move(client), argv.at(0), argv.at(1));
+}
+
+void ListJobs(
+    google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+        client,
+    std::vector<std::string> const& argv) {
+  //! [storage_batch_list_jobs]
+  [](google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+         client,
+     std::string const& parent) {
+    for (auto const& job : client.ListJobs(parent)) {
+      if (!job) throw job.status();
+      std::cout << job->DebugString() << "\n";
+    }
+  }
+  //! [storage_batch_list_jobs]
+  (std::move(client), argv.at(0));
+}
+
+void GetJob(
+    google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+        client,
+    std::vector<std::string> const& argv) {
+  //! [storage_batch_get_job]
+  [](google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+         client,
+     std::string const& name) {
+    auto job = client.GetJob(name);
+    if (!job) throw job.status();
+    std::cout << "Got job: " << job->DebugString() << "\n";
+  }
+  //! [storage_batch_get_job]
+  (std::move(client), argv.at(0));
+}
+
+void CancelJob(
+    google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+        client,
+    std::vector<std::string> const& argv) {
+  //! [storage_batch_cancel_job]
+  [](google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+         client,
+     std::string const& name) {
+    auto response = client.CancelJob(name);
+    if (!response) throw response.status();
+    std::cout << "Cancelled job: " << response->DebugString() << "\n";
+  }
+  //! [storage_batch_cancel_job]
+  (std::move(client), argv.at(0));
+}
+
+void DeleteJob(
+    google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+        client,
+    std::vector<std::string> const& argv) {
+  //! [storage_batch_delete_job]
+  [](google::cloud::storagebatchoperations_v1::StorageBatchOperationsClient
+         client,
+     std::string const& name) {
+    auto status = client.DeleteJob(name);
+    if (!status.ok()) throw status;
+    std::cout << "Deleted job\n";
+  }
+  //! [storage_batch_delete_job]
+  (std::move(client), argv.at(0));
+}
+
+void AutoRun(std::vector<std::string> const& argv) {
+  if (!argv.empty()) throw google::cloud::testing_util::Usage{"auto"};
+  google::cloud::testing_util::CheckEnvironmentVariablesAreSet(
```

Comment:
I don't think we want all these new environment variables. More than likely you can leverage random strings for some of these.